### PR TITLE
read 2022 IEA Data

### PR DESCRIPTION
**Pull request by fbenke-pik**

### CODE REVIEW

File: R/calcIOEdgeBuildings.R [hunk 1]
```diff
@@ -36,7 +36,10 @@ calcIOEdgeBuildings <- function(subtype = c("output_EDGE", "output_EDGE_building
   # delete NAs rows
   ieamatch <- ieamatch[c("iea_product", "iea_flows", target, "Weight")] %>%
     na.omit() %>%
-    unite("target", all_of(target), sep = ".")
```
Comment (LaviniaBaumstark) [inline R/calcIOEdgeBuildings.R:39]: this is only working with the new IEA data, right? Is there a version number which could be increased to indicate this?

File: R/readIEA.R [hunk 2]
```diff
@@ -11,27 +11,61 @@
...
+
+    energyBalancesFile <- "IEA-Energy-Balances-2022/worldbig.csv"
+    data <- fread(
+      file = energyBalancesFile,
+      col.names = c("COUNTRY", "PRODUCT", "FLOW", "TIME", "ktoe"),
+      colClasses = c("character", "character", "character", "numeric", "character"),
+      sep = ";", stringsAsFactors = FALSE, na.strings = c("x", "..", "c"), skip = 2, showProgress = FALSE
+    )
+
+    customMapping <- c(
```
Comment (LaviniaBaumstark) [inline R/readIEA.R:29]: I think, this should be added to the mapping country2ISO in madrat

**Review by LaviniaBaumstark — CHANGES_REQUESTED** [review]

File: R/calcIOEdgeBuildings.R (hunk 1 above)
Comment (fbenke-pik) [reply R/calcIOEdgeBuildings.R:39]: We are explicitly reading in new IEA data, so this only has to work with new IEA data, right?
https://github.com/fbenke-pik/mrcommons/blob/IEA/R/readIEA.R#L21

Comment (LaviniaBaumstark) [reply R/calcIOEdgeBuildings.R:39]: yes, right, it is all in one PR but if we anyway have a version number, this would ensure consistency too

Comment (fbenke-pik) [reply R/calcIOEdgeBuildings.R:39]: @robinhasse once this mrrcommons branch is merged and we have a version number, you might have to update the required version of mrcommons here: https://github.com/pik-piam/mredgebuildings/blob/main/DESCRIPTION

Comment (fbenke-pik) [reply R/calcIOEdgeBuildings.R:39]: done

File: R/readIEA.R (hunk 2 above)
Comment (fbenke-pik) [reply R/readIEA.R]: done

**Review by LaviniaBaumstark — COMMENTED** [review]
nice, but checks are failing

**Comment by fbenke-pik** [comment]
doesn't look like this is a problem with the package, but with the lucode library.

```
Run lucode2:::validkey(stopIfInvalid = TRUE)
Error: .onLoad failed in loadNamespace() for 'styler', details:
  call: loadNamespace(j <- i[[1L]], c(lib.loc, .libPaths()), versionCheck = vI[[j]])
  error: there is no package called ‘R.methodsS3’
Execution halted
Error: Process completed with exit code 1.
```

File: R/calcIOEdgeBuildings.R (hunk 1 above)
Comment (robinhasse) [reply R/calcIOEdgeBuildings.R:39]: Thanks for the hint. This is still under development and not in use currently but still good that you reminded me.

**Review by robinhasse — APPROVED** [review]
Looks good to me!

**Review by LaviniaBaumstark — APPROVED** [review]

**Pull request merged**